Code/GeorgiaLibrarySystem-/GTLService/DataAccess/Code/MaterialDa_Code.cs

```diff
@@ -0,0 +1,115 @@
+﻿using System.Collections.Generic;
+using System.Data.SqlClient;
+using System.Linq;
+using Core;
+
+namespace GTLService.DataAccess.Code
+{
+    public class MaterialDa_Code
+    {
+        private readonly Context _context;
+        public MaterialDa_Code(Context context)
+        {
+            this._context = context;
+        }
+
+        public virtual List<Material> ReadMaterials(int isbn, string title, string author, int numOfRecords)
+        {
+            return _context.Materials
+                .Where(x => (isbn == 0 || x.ISBN.Equals(isbn.ToString())) &&
+                            x.Author.Contains(author) &&
+                            x.Title.Contains(title)
+                )
+                .Take(numOfRecords)
+                .ToList();
+        }
+
+        public virtual List<Copy> ReadCopies(int isbn, string typeName)
+        {
+            var a = _context.Copies
+                .Where(x => (isbn == 0 || x.ISBN.Equals(isbn.ToString())) &&
+                    (typeName.Contains("0") || x.TypeName.Contains(typeName))
+                )
+                .ToList();
+            return a;
+        }
+
+        public virtual readAllMaterial ReadMaterials(int isbn)
+        {
+            return _context.readAllMaterials.Find(isbn);
+        }
+
+        public virtual bool CheckMaterialIsbn(int isbn)
+        {
+            var material = _context.Materials.Find(isbn);
+            return material != null;
+        }
+
+        public virtual bool CheckCopyId(int id)
```

Comment:
why does this material da deal with copies? shouldnt there be a new class for them (like this its harder to understand as alot of stuff in one class and leter on when expending we wont know in which class we could look for the method we need and have code duplication)